Pd 1011 update core to scale migration articles (#2499)
truenas/documentation: content/SCALE/GettingStarted/Migrate/MigratePrep.md

- migrate
---

{{< include file="/_includes/MigrateCOREtoSCALEWarning.md" >}}

## What can or cannot migrate?

{{< include file="/_includes/COREMigratesList.md" >}}

## Preparing for Migration
Before you attempt to migrate your CORE system to a SCALE major version. 

> {{< hint type="warning" title="Using USB Devices" >}}

Author: This was in the context as the medium for backups. I've clarified this content in the next commit that has suggested PR changes.

We strongly recommend not using USB flash drives or USB-attached drives as these can have issues, including with recovering backed up files.
For more information on using USB drives and devices read the [CORE Hardware Guide]({{< relref "/CORE/GettingStarted/CoreHardwareGuide.md" >}}).
If you must use a USB type device, verify you can access files on the device before you upgrade/migrate to SCALE.
{{< /hint >}}
{{< enterprise >}}
CORE Enterprise customers are encouraged to contact Support for assistance with the process of moving from CORE to SCALE, especially customers with HA systems.
{{< /enterprise >}}

1. Upgrade your CORE system to the most recent publicly-available CORE major maintenace release version.
   TrueNAS systems on 12.0x or earlier should upgrade to the latest CORE 13.0 release (e.g 13.0-U6.1 or newer) prior to migrating to SCALE.
   CORE systems at the latest 13.0 release can use the [iso upgrade](#migrating-using-an-iso-file-to-upgrade) method to migrate to SCALE.

>    Migrate [GELI-encrypted pools](https://www.truenas.com/docs/core/coretutorials/storage/pools/storageencryption/#geli-pool-migrations) to a non-GELI-encrypted pool before upgrading from CORE 12.0x or earlier releases!

Author: Added as step 2, then renumbered the rest of the list.

   
   <input type="checkbox"> Deprecated services - Record the settings for [services deprecated in SCALE](#migrating-from-deprecated-services).
   
   <input type="checkbox"> VMs - If you have virtual machines configured in CORE, write down or screenshot network and other setting information.
   
   <input type="checkbox"> Plugins or jails - If you use plugins and jails, they do not migrate. Record settings for each plugin/jail and back up the data associated with each.
   
   <input type="checkbox"> CAs, certificates, CSRs - If you added certificate authorities, certificates or certificate signing requests to CORE, these should migrate with the system config file, but as a precaution against possible malformed certificates copy or screenshot all CA, certificate, and CSR settings, and save the public and private keys of each. Make sure you have backed-up copies of certificates used in CORE to import or configure in SCALE.
   
   <input type="checkbox"> Usernames beginning with (0-9) - Review local user account names and rename or replace these with a letter or underscore before migrating.
   When you set up the first required pool on SCALE import this pool first.

>    <input type="checkbox"> Tunables on CORE - SCALE does not use **Tunables** the way CORE does. SCALE provides script configuration on the **System Settings > Advanced** screen as **Sysctl** scripts. A future release of SCALE could introduce similar tunables options found in CORE but for now it is not available.

Reviewer: Don't think we want to promise future functionality that hasn't necessarily been scheduled or confirmed yet.
```suggestion
   <input type="checkbox"> Tunables on CORE - SCALE does not use **Tunables** the way CORE does. SCALE provides script configuration on the **System Settings > Advanced** screen as **Sysctl** scripts.
```

Author: This was in the article already. I'll reword this.

   <input type="checkbox"> Global self-encrypting drive (SED) Password - Unlock these drives in CORE before you clean install SCALE. Note the SED password configured in CORE to use in SCALE.

   <input type="checkbox"> Credentials - Note the credentials for SSH connections and keypairs, and cloud service backup providers configured in CORE if you do not have them recorded in other files kept secured outside of CORE.

   <input type="checkbox"> Data protection tasks - Record or screenshot replication, periodic snapshot, cloud sync, or other tasks settings to reconfigure these in SCALE if you want to duplicate these tasks. 

4. Write down your network configuration information or take screenshots of the global network settings, interfaces, static IP addresses, and aliases configured.
   
   FreeBSD and Linux use different nomenclature for network interfaces, bridges, LAGGs, and VLANs. 
   Because of the difference, network settings can either get lost or not transfer which means you have no network connectivity.
   You can find interface names in the CORE UI on the **Network > Interfaces** screen.

>    If using TrueNAS System servers, to find the Linux interface ID equivalent, refer to the network port ID manuals for your model found in the [Solutions > TrueNAS Systems](https://www.truenas.com/docs/hardware/) area of the Documentation Hub for network port assignments.   

Reviewer: ```suggestion
   When using a TrueNAS Enterprise system from iXsystems, refer to the network port ID manuals of your [TrueNAS Systems](https://www.truenas.com/docs/hardware/) to find the network port assignments in TrueNAS SCALE.
```

>    If not using TrueNAS system hardware, refer to the manual or documentation provided with your system or locate this information on your server hardware and take note of it.

Reviewer: ```suggestion
   When using custom hardware for TrueNAS, refer to the manual or documentation provided with your system or locate this information on your server hardware and take note of it.
```

   If there are issues after a clean install of SCALE from an <file>iso</file> file or you are not using DHCP for network and interface configuration, use the information from your CORE settings to configure your SCALE network settings and to reconfigure your static IPs or aliases.
      {{< include file="/_includes/NetworkInstallRequirementsSCALE.md" >}}

5. Migrate the deprecated S3 MinIO service (if in use). See [services deprecated in SCALE](#migrating-from-deprecated-services).
   Read and follow instructions in [Migrating from MinIO S3](https://www.truenas.com/docs/scale/22.12/scaletutorials/apps/communityapps/minioclustersscale/migratingfroms3service/)!
   Make sure S3 MinIO data is backed up as a precaution. The S3 service requires installing SCALE 22.12.3 (Bluefin).
   This version of SCALE provides access to both the S3 service and the MinIO app you migrate to. 

6. Back up your stored data files.
   If you need to do a clean install with the SCALE <file>iso</file> file, you can import your data pools into SCALE.

>    Verify you can access your backed up filed before you upgrade/migrate to SCALE.

Reviewer: ```suggestion
   Verify you can access your backed up files before you upgrade/migrate to SCALE.
```

> 7. Back up any critical data. 

Reviewer: 6. and 7. are essentially the same thing, go ahead and combine.

Author: Combined in the next commit.
